### PR TITLE
Scroll position reset when routing

### PR DESCRIPTION
**Pull request by frohikey**
Issue: #3

Vue router doesn't handle scroll positioning automatically. 
You can use this, but it does work only when you're in history mode and windows.scrollTo "works" (it depends on CSS)

```javascript
const router = new VueRouter({
  routes, // short for routes: routes
  linkActiveClass: 'active',
  mode: 'history'
  scrollBehavior (to, from, savedPosition) {    
    if (savedPosition) {
       return savedPosition
     } else {
       return { x: 0, y: 0 }
     }
   }
})
```

Unfortunately it doesn't work in our case. The reason is - side panel and main panel both have ``overflow: auto;`` and thus ``window.scrollTo`` does nothing.

I've fixed it by adding ``vue-scrollto`` plugin and resetting a scroll position when routing.

### CODE REVIEW

**Comment by cristijora** [comment]
Thanks a lot for the idea and the PR. 
I have a couple of questions here though:
By placing this in router.beforeEach does it mean that if you go back from browser back button scroll position will not be saved?
Also it would be good to move the plugin to `dependencies`

File: package.json [hunk 1]
```diff
@@ -83,7 +83,8 @@
     "webpack-bundle-analyzer": "^2.2.1",
     "webpack-dev-middleware": "^1.10.0",
     "webpack-hot-middleware": "^2.16.1",
-    "webpack-merge": "^2.6.1"
+    "webpack-merge": "^2.6.1",
+    "vue-scrollto": "^2.6.9"
```
Comment (cristijora) [inline package.json]: Would be better to move this to dependencies instead of devDependencies.

**Comment by frohikey** [comment]
Yeah, that's a problem. Position is not saved.
Imho it could be solved only by using "history" mode in router. And I didn't want to make such a big change.
I will think about it. Take it as an idea atm :)

**Comment by cristijora** [comment]
Well history mode is good, but there are some disatvantages to it because it requires a server to run and you can no longer host a demo on gh-pages or maybe you can but with some effort so don't think that's an option

**Comment by cristijora** [comment]
I am going to reject this because this can be fixed with some css and vue-router in history mode
See last comment from #3. Thanks for the PR anyway. Not sure I would implement this but rather fix the css and add in the documentation that you can use `history` mode and scroll to top on each route change

**Pull request closed without merging**